frontend/src/metabase/common/components/FilterPicker/FilterPicker.tsx

```diff
@@ -121,13 +124,18 @@ export function FilterPicker({
   const FilterWidget = getFilterWidget(column);
 
   if (FilterWidget) {
+    const isSameColumn =
```

Comment:
I wouldn't do that. I'd say that if you choose another column, then drop the existing filter clause in the state. This way we won't need to compare columns

Reply:
@ranquild do you suggest adding something like that (because `filter` is just a prop now)?

```tsx
function FilterPicker({ filter: initialFilter }) {
  const [filter, setFilter] = useState(initialFilter)
}
```

If so, I'm worried it'd make the following scenario awkward:

1. Click a filter like "Category is Gadget"
2. Go back to the list of columns, we reset the `filter` state
3. Open the Category column again
4. Now we need to find the original `filter` and put it back in state

We'd also most likely need a `useEffect` to keep the `filter` prop and state in sync which is rarely great

WDYT?

Reply:
Yes, but after 3 we don't need to find the original `filter`. We need to drop it. We follow this pattern in the `DatePicker` when switching between specific/relative/exclude dates

Reply:
Even for exclude dates, if you go back to the list of temporal buckets and select the same one, your value would be reset

Reply:
Oh, I didn't notice that. Are you sure that's an expected behavior and not a bug in the original relative date picker?

Reply:
Not sure, but it made everything simpler. We could revise this later. If we do revise this, we could do it without `isSameColumn` I think, but we would need to be smarter with our state.

Reply:
@kulyk I just checked - our current FilterPicker doesn't preserve the filter clause either

Reply:
@ranquild I've reworked it, please take another look